tutorials/2023-11-06-offchain-sdk-ts-generation-with-cpm/index.md

```diff
@@ -0,0 +1,109 @@
+    hosts:
+      - 'http://127.0.0.1:10332'
+
+```
+
+Then just run `cpm run` on the terminal and the SDK should be generated at the `cpm/offchain/ts/neotoken` folder. Check CPM's [documentation](https://github.com/CityOfZion/cpm/blob/master/docs/config.md) to see other options and how to better configure this `cpm.yaml` file.
+### With Command Prompt
+To generate a SDK using CPM on the command line, first, you'll need to have the contract manifest, you can get it by running: 
+```
+cpm download manifest -c 0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5 -N https://mainnet1.neo.coz.io:443
+```
+Then you'll need to run the generate command:
+```
+cpm generate ts -m .\contract.manifest.json -c 0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5
+```
+And the SDK will be generated at the `cpm/offchain/ts/neotoken` folder.
+## 3. Initializing and running the SDK
+Now that the SDK is already generated, all you need to do is import it in your project and initialize it to be able to call it.
+You'll only need to instantiate a class that implements `Neo3-Invoker`, like [`NeonInvoker`](https://github.com/CityOfZion/neon-dappkit/blob/main/packages/neon-dappkit/src/NeonInvoker.ts) or [`WalletConnect SDK`](https://github.com/CityOfZion/wallet-connect-sdk/tree/main).
+```ts
+import { NeoToken } from './cpm_out/offchain/ts/neotoken'
+
+import { NeonInvoker } from "@cityofzion/neon-dappkit"
+import WcSdk from '@cityofzion/wallet-connect-sdk-core'
+import SignClient from '@walletconnect/sign-client'
+
+async function createNeoTokenSDK(): Promise<NeoToken> {
+	return new NeoToken({
+		scriptHash: NeoToken.SCRIPT_HASH,
+		// Alternatively, use WcSDK instead of NeonInvoker
+		invoker: await NeonInvoker.init({
+			rpcAddress: 'https://mainnet1.neo.coz.io:443',
+		})
+	})
+}
+
+async function createNeoTokenSDKWithWcSDK(): Promise<NeoToken> {
+	let wcSdk
+	let signClient
+
+	signClient = await SignClient.init({
+        projectId: 'a9ff54e3d56a52230ed8767db4d4a810', // the ID of your project on Wallet Connect website
+        relayUrl: 'wss://relay.walletconnect.com', // we are using walletconnect's official relay server
+        metadata: {
+          name: 'MyApplicationName', // your application name to be displayed on the wallet
+          description: 'My Application description', // description to be shown on the wallet
+          url: 'https://myapplicationdescription.app/', // url to be linked on the wallet
+          icons: ['https://myapplicationdescription.app/myappicon.png'], // icon to be shown on the wallet
+        },
+    })
+    wcSdk = new WcSdk(signClient)
+
+    return new NeoToken({
+		scriptHash: NeoToken.SCRIPT_HASH,
+		invoker: wcSdk
+	})
+}
+
+```
+
+Then you can invoke the contract methods with this SDK:
+```ts
+async main() {
+	const neoToken = await createNeoTokenSDKWithNeonDappkit()
+	console.log(await neoToken.symbol())        // NEO
+	console.log(await neoToken.decimals())      // 0
+	console.log(await neoToken.totalSupply())   // 100000000
+}
+```
```

Comment:
It's important to explain the difference between using NeonInvoker and WcSdk and why would you choose one over the other, the different situations where you need one over the other.